### PR TITLE
Add jsxmlserializer

### PR DESCRIPTION
**Pull request by juancarlospaco**
See https://github.com/nim-lang/Nim/pull/16339#issue-538769691 and  https://github.com/nim-lang/Nim/pull/16339#issuecomment-744485709 and https://github.com/nim-lang/fusion/pull/57#issue-539766026 and https://github.com/nim-lang/fusion/pull/56#issue-539747803 and https://github.com/nim-lang/fusion/pull/55#issue-539726540
and https://github.com/nim-lang/fusion/pull/58#issue-539886249

### CODE REVIEW

File: src/fusion/js/jsxmlserializer.nim [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+type XMLSerializer* = ref object of JsRoot  ## XMLSerializer API.
+
+func newXMLSerializer*(): XMLSerializer {.importjs: "new XMLSerializer()".}
+  ## https://developer.mozilla.org/en-US/docs/Web/API/XMLSerializer
+
+func serializeToString*(this: XMLSerializer; node: Node): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/API/XMLSerializer/serializeToString
+
+func serializeToString*(node: Node): cstring {.importjs:
```
Comment (timotheecour) [inline src/fusion/js/jsxmlserializer.nim:16]: can't this be:
```nim
func serializeToString*(node: Node): cstring =
  ## ...
  newXMLSerializer(). serializeToString(node)
```

File: src/fusion/js/jsxmlserializer.nim [hunk 2]
```diff
@@ -0,0 +1,25 @@
+## - `XMLSerializer` for the JavaScript target: https://developer.mozilla.org/en-US/docs/Web/API/XMLSerializer
+when not defined(js) and not defined(Nimdoc):
```
Comment (timotheecour) [inline src/fusion/js/jsxmlserializer.nim:2]: nit: use `nimdoc` (even if equivalent, it's more common hence less gotcha's say during search)

File: src/fusion/js/jsxmlserializer.nim [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  {.fatal: "Module jsxmlserializer is designed to be used with the JavaScript backend.".}
+
+from dom import Node
+export Node
```
Comment (timotheecour) [inline src/fusion/js/jsxmlserializer.nim:6]: do we need to re-export?
no strong opinion if you have a good reason

**Review by timotheecour — APPROVED** [review]
LGTM after addressing minor review points

**Review by timotheecour — APPROVED** [review]

File: src/fusion/js/jsxmlserializer.nim (hunk 3 above)
Comment (timotheecour) [reply src/fusion/js/jsxmlserializer.nim:6]: ping on this

**Review by timotheecour — APPROVED** [review]

**Comment by timotheecour** [comment]
@juancarlospaco LGTM; can you ask for another LGTM so we can merge this?

File: src/fusion/js/jsxmlserializer.nim [hunk 4]
```diff
@@ -0,0 +1,20 @@
+## - `XMLSerializer` for the JavaScript target: https://developer.mozilla.org/en-US/docs/Web/API/XMLSerializer
+when not defined(js) and not defined(nimdoc):
+  {.fatal: "Module jsxmlserializer is designed to be used with the JavaScript backend.".}
+
+from dom import Node
```
Comment (xflywind) [inline src/fusion/js/jsxmlserializer.nim:5]: nit: use std/dom

File: src/fusion/js/jsxmlserializer.nim [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+func newXMLSerializer*(): XMLSerializer {.importjs: "new XMLSerializer()".}
+  ## https://developer.mozilla.org/en-US/docs/Web/API/XMLSerializer
+
+func serializeToString*(this: XMLSerializer; node: Node): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/API/XMLSerializer/serializeToString
+
+
+runnableExamples:
+  from dom import document
```
Comment (xflywind) [inline src/fusion/js/jsxmlserializer.nim:17]: ditto

**Review by xflywind — APPROVED** [review]

**Review by timotheecour — APPROVED** [review]

**Pull request merged**